SourceX/DiabloUI/selyesno.cpp

```diff
@@ -24,18 +24,16 @@ void selyesno_Free()
 		UiListItem* pUIListItem = vecSelYesNoDialogItems[i];
 		if(pUIListItem)
 			delete pUIListItem;
-
-		vecSelYesNoDialogItems.clear();
 	}
+	vecSelYesNoDialogItems.clear();
 
 	for(int i = 0; i < (int)vecSelYesNoDialog.size(); i++)

```

Comment:
There are some weird casts here. Perhaps use a `size_t` index?

```c++
for (std::size_t i = 0; i < vecSelYesNoDialog.size(); ++i)
```

(or `size_t` instead of `std::size_t` if the XBox compiler doesn't like it)

Reply:
`std::size_t` was used in a few other places so I made the change to all the remaning loops as well as some other cleanups and a partial fix to the list not being clickable.